examples/granger_causality.py

```diff
@@ -0,0 +1,405 @@
+
+import mne
+from mne.datasets.fieldtrip_cmc import data_path
+from mne_connectivity import spectral_connectivity_epochs
+
+###############################################################################
+# Background
+# ----------
+#
+# Multivariate forms of signal analysis allow you to simultaneously consider
+# the activity of multiple signals. In the case of connectivity, the
+# interaction between multiple sensors can be analysed at once, producing a
+# single connectivity spectrum. This approach brings not only practical
+# benefits (e.g. easier interpretability of results from the dimensionality
+# reduction), but can also offer methodological improvements (e.g. enhanced
+# signal-to-noise ratio and reduced bias).
+#
+# Additionally, it can be of interest to examine the directionality of
+# connectivity between signals, providing additional clarity to how information
+# flows in a system. One such directed measure of connectivity is Granger
+# causality (GC). A signal, :math:`\boldsymbol{x}`, is said to Granger-cause
+# another signal, :math:`\boldsymbol{y}`, if information from the past of
+# :math:`\boldsymbol{x}` improves the prediction of the present of
+# :math:`\boldsymbol{y}` over the case where only information from the past of
+# :math:`\boldsymbol{y}` is used. Note: GC does not make any assertions about
+# the true causality between signals.
+#
+# The degree to which :math:`\boldsymbol{x}` and :math:`\boldsymbol{y}` can be
+# used to predict one another can be quantified using vector autoregressive
+# (VAR) models. Considering the simpler case of time domain connectivity, the
+# VAR models are as follows:
+#
+# :math:`y_t = \sum_{k=1}^{K} a_k y_{t-k} + \xi_t^y` ,
+# :math:`Var(\xi_t^y) := \Sigma_y` ,
+#
+# and :math:`\boldsymbol{z}_t = \sum_{k=1}^K \boldsymbol{A}_k
+# \boldsymbol{z}_{t-k} + \boldsymbol{\epsilon}_t` ,
+# :math:`\boldsymbol{\Sigma} := \langle \boldsymbol{\epsilon}_t
+# \boldsymbol{\epsilon}_t^T \rangle = \begin{bmatrix} \Sigma_{xx} & \Sigma_{xy}
+# \\ \Sigma_{yx} & \Sigma_{yy} \end{bmatrix}` ,
+#
+# representing the reduced and full VAR models, respectively, where: :math:`K`
+# is the order of the VAR model, determining the number of lags, :math:`k`,
+# used; :math:`\boldsymbol{Z} := \begin{bmatrix} \boldsymbol{x} \\
+# \boldsymbol{y} \end{bmatrix}`; and :math:`\xi` and
+# :math:`\boldsymbol{\epsilon}` are the residuals of the VAR models. In this
+# way, the information of the signals at time :math:`t` can be represented as a
+# weighted form of the information from the previous timepoints, plus some
+# residual information not encoded in the signals' past. In practice, VAR model
+# parameters are computed from an autocovariance sequence generated from the
+# time-series data using the Yule-Walker equations :footcite:`Whittle1963`.
+#
+# By comparing the residuals, or errors, of the reduced and full VAR models, we
+# can therefore estimate how much :math:`\boldsymbol{x}` Granger-causes
+# :math:`\boldsymbol{y}`:
```

Comment:
Sort of. It would be more the variance of the residuals of the (reduced) model based on the information in y alone vs. the variance of the residuals of the (full) model based on the information in both y and x. I have re-written this part to try and make it clearer.